gui/default/untrusted/syncthing/core/syncthingController.js

```diff
@@ -1433,40 +1433,36 @@ angular.module('syncthing.core')
                 }
             }
             $scope.currentDevice._addressesStr = deviceCfg.addresses.join(', ');
-
             initShareEditing('device');
-            for (var folderID in $scope.folders) {
-                var found = false;
-                for (var i = 0; i < $scope.folders[folderID].devices.length; i++) {
-                    if ($scope.folders[folderID].devices[i].deviceID === deviceCfg.deviceID) {
-                        found = true;
+            $scope.deviceFolders($scope.currentDevice).forEach(function (folderID) {
+                $scope.currentSharing.shared.push($scope.folders[folderID]);
+                $scope.currentSharing.selected[folderID] = true;
+                var folderdevices = $scope.folders[folderID].devices;
+                for (var i = 0; i < folderdevices.length; i++) {
+                    if (folderdevices[i].deviceID === deviceCfg.deviceID) {
+                        $scope.currentSharing.encryptionPasswords[folderID] = folderdevices[i].encryptionPassword;
                         break;
                     }
                 }
-                if (found) {
-                    $scope.currentSharing.encryptionPasswords[folderID] = $scope.folders[folderID].devices[i].encryptionPassword;
-                    $scope.currentSharing.shared.push($scope.folders[folderID]);
-                } else {
-                    $scope.currentSharing.unrelated.push($scope.folders[folderID]);
-                }
-                $scope.currentSharing.selected[folderID] = found;
-            }
-
+            });
+            $scope.currentSharing.unrelated = $scope.folderList().filter(function (n) {
+                return !$scope.currentSharing.selected[n.id];
+            });
```

Comment:
As far as I understand this does not change behaviour. What's the reasoning for this change?

Reply:
As I said, just trying to keep the diff between default and untrusted as small as possible, without having two different implementations of the same algorithm.  I picked the one from default because it was easier to grasp for me at least.